src/Compilers/Core/Portable/CommandLine/CommonCompiler.CompilerEmitStreamProvider.cs

```diff
@@ -40,15 +40,71 @@ public override Stream CreateStream(DiagnosticBag diagnostics)
 
                 try
                 {
-                    return _streamToDispose = _compiler.FileOpen(_filePath, PortableShim.FileMode.Create, PortableShim.FileAccess.ReadWrite, PortableShim.FileShare.None);
+                    try
+                    {
+                        return OpenFileStream();
+                    }
+                    catch (IOException e)
+                    {
+                        // Other process is reading the file preventing us to write to it.
+                        // We attempt to rename and delete the file in case the reader opened it with FileShare.Delete flag that
+                        // allows the file to be deleted by other processes.
+                        //
+                        // Note that if the file is marked "readonly" or the current user doesn't have sufficient privileges
+                        // the exception thrown is UnauthorizedAccessException, not IOException, so we won't attempt to delete the file.
+
+                        try
+                        {
+                            const int eWin32SharingViolation = unchecked((int)0x80070020);
+
+                            if (PathUtilities.IsUnixLikePlatform)
+                            {
+                                // Unix & Mac are simple: just delete the file in the directory. 
+                                // The memory mapped content remains available for the reader.
+                                PortableShim.File.Delete(_filePath);
```

Comment:
Or is there any value in failing the build in case someone else has the output files open? Every time I see "file is being used by another process" during build it's a hindrance not help. And it inflicts pain on all tools that need to do various forms of shadow-copying (like xunit, interactive, etc.)